io/src/main/scala/sbt/io/IO.scala

```diff
@@ -1197,4 +1172,67 @@ object IO {
    */
   def copyModifiedTime(fromFile: File, toFile: File): Unit =
     Milli.copyModifiedTime(fromFile, toFile)
+
+  /**
+   * Return the last modification timestamp of the specified file,
+   * in milliseconds since the Unix epoch (January 1, 1970 UTC).
+   * Please see IO.getModifiedTime() for further information.
+   *
+   * In contrast to getModifiedTime(), if the specified file does
+   * not exist, this method will return 0L, rather than throwing
+   * a FileNotFoundException.
+   *
+   * @see getModifiedTime
+   */
+  def lastModified(file: File): Long =
```

Comment:
I think it's too subtle/confusing to have different variety of `getModifiedTime` that differs in similar sounding name.

My suggestion here would be to have two versions:
- `getModifiedTimeTry` that returns `scala.util.Try`.
- `getModifiedTime` that returns `Long` and uses `0L` when the file doesn't exist.

Reply:
`IO.lastModified` and `IO.setLastModified` have the exact same semantics as `f.lastModified` and `f.setLastModified`, which is why retaining the same names is important.
The other methods are just ancillary, and after these patches they will be mostly unused within sbt.
Actually, we have spent a great deal of time already on what should have been a minor matter, so I'd avoid further tweaks. At this point I would strongly recommend that we just merge these pull requests, and move forward with other topics.

Reply:
Any public method on `IO` becomes semi-permanent API of sbt, so if we can avoid creating confusing duplicates that will save our headaches in the future. Since io 1.1.2 is already out with `IO.getModifiedTime(..)` we are committed with the name. For the same reason we are going through this exercise, anyone who tries to replace the usage of `File#lastModified` should use the return-`0L` version.

Given all other modules are already calling `IO.getModifiedTime(..)`, changing here also saves changing the code downstream.

Reply:
Since `IO.getModifiedTime` is already included in `IO`, we have to keep respecting the specification described in the Scaladoc. It is already described as a method that throws exception, so we cannot change it again now.

If you are concerned about too many time-related methods being exposed in `IO`, it is perfectly ok to only expose `xxxModifiedTime`, keeping the `xxxlastModified` calls internal. I can just place them in `internal.io.Milli`. However, we cannot change `IO.getModifiedTime` again, now.

Consequently, the only possible course of action is transitioning the downstream code to (differently named) calls that go back to the `0L` semantics. That is what these pull requests do, upon @dwijnand's suggestion.

It is convenient to name them in the same way as the `java.util.File` ones, for clarity. It is true however that they do not need to be exposed in `IO` at all. (Incidentally, you are the one who asked for `getModifiedTime` to be exposed in `IO`)

So, as a middle ground solution, I propose that I remove `lastModified` and `setLastModified` from `IO` and move them to `internal.io.Milli`, changing downstream accordingly.

Reply:
Given that sbt 1.1.0 hasn't reached final, I am personally ok with `IO.getModifiedTime` changing the semantics from the previous description in Scaladoc.

I am fine with either:
- Change to return `IO.getModifiedTime` `0L`, or
- Make `xxxLastModified` internal.

@dwijnand What do you think?

Reply:
Given this is sbt/io and not sbt I'm of the opinion that we can't change semantics between stable releases (aka from 1.1.2 to 1.1.3).

What's a good alternative name to `getModifiedTime` to put on `IO` that means "last modified time or 0`. `getLastModifiedOrZero`? `getLastModifiedSafe`?

Personally I'm fine with `lastModified`, but I'd be equally happy if a better name is found that's distinct enough from `getModifiedTime` to avoid confusion.

Reply:
How about we deprecate `IO.getModifiedTime` and add:

1. `getLastModifiedTry` that uses `Try`
2. `getLastModifiedOrZero` that returns `0L`

?

Reply:
Fine by me.

Reply:
If it's deprecated then it's effectively out of the API.
`Try` should be the replacement for someone who wants throwy behavior.

Reply:
Maybe instead of a `getLastModifiedTry` it should be `getLastModifiedUnsafe` which throws exceptions.